ensemble/async/connection/kafka/kafka.go

```diff
@@ -13,17 +13,10 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package async
+package kafka
 
-// KafkaConnection represents a Kafka broker connection settings.
-type KafkaConnection struct {
-	bootstrapServers string
-}
-
-// AmazonServiceConnection represents an Amazon service connection settings.
-type AmazonServiceConnection struct {
-	region           string
-	endpointOverride string
-	accessKey        string
-	secretKey        string
+// Connection represents broker connection settings.
+type Connection struct {
+	// BootstrapServers represents the list of bootstrap servers.
+	BootstrapServers string
```

Comment:
Could you explain the rationale for splitting the file into two and putting those in dedicated packages? Is it a matter of naming? Of Go conventions or best practices?

Reply:
Yes in fact, this is the way in go to have "namespace" and separated concerns.